pkg/controller/plan/adapter/ocp/builder_test.go

```diff
@@ -0,0 +1,367 @@
+	}
+}
+
+func TestFindVMInManifestItems_EmptyItems(t *testing.T) {
+	t.Parallel()
+
+	// Create empty items slice
+	items := []runtime.RawExtension{}
+
+	// Create builder and test findVMInManifestItems
+	builder := createTestBuilder()
+
+	testScheme := createTestScheme()
+	decode := serializer.NewCodecFactory(testScheme).UniversalDeserializer().Decode
+	foundVM := builder.findVMInManifestItems(items, decode)
+
+	// Assertions
+	if foundVM != nil {
+		t.Errorf("Expected nil when no items present, got %v", foundVM)
+	}
+}
+
+func TestFindVMInManifestItems_NoVMInItems(t *testing.T) {
+	t.Parallel()
+
+	// Create a non-VM object (ConfigMap)
+	cm := &core.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-cm",
+			Namespace: "test-namespace",
+		},
+		Data: map[string]string{
+			"key": "value",
+		},
+	}
+
+	// Serialize the ConfigMap to JSON
+	cmBytes, err := json.Marshal(cm)
+	if err != nil {
+		t.Fatalf("Failed to marshal ConfigMap: %v", err)
+	}
+
+	// Create items slice with non-VM object
+	items := []runtime.RawExtension{
+		{
+			Raw: cmBytes,
+		},
+	}
+
+	// Create builder and test findVMInManifestItems
+	builder := createTestBuilder()
+
+	testScheme := createTestScheme()
+	decode := serializer.NewCodecFactory(testScheme).UniversalDeserializer().Decode
+	foundVM := builder.findVMInManifestItems(items, decode)
+
+	// Assertions
+	if foundVM != nil {
+		t.Errorf("Expected nil when no VM in items, got %v", foundVM)
+	}
+}
+
+func TestFindVMInManifestItems_InvalidJSON(t *testing.T) {
+	t.Parallel()
+
+	// Create items slice with invalid JSON
+	items := []runtime.RawExtension{
+		{
+			Raw: []byte(`{"invalid": json}`), // Invalid JSON
+		},
+	}
+
+	// Create builder and test findVMInManifestItems
+	builder := createTestBuilder()
+
+	testScheme := createTestScheme()
+	decode := serializer.NewCodecFactory(testScheme).UniversalDeserializer().Decode
+	foundVM := builder.findVMInManifestItems(items, decode)
+
+	// Assertions - should handle errors gracefully and continue searching
+	if foundVM != nil {
+		t.Errorf("Expected nil when invalid JSON present, got %v", foundVM)
+	}
+}
+
+func TestBuilder_MacAddressPreservation(t *testing.T) {
+	t.Parallel()
+
+	// Test that the builder preserves MAC addresses from source VM through DeepCopy
+	sourceVm := &cnv.VirtualMachine{
+		Spec: cnv.VirtualMachineSpec{
+			Template: &cnv.VirtualMachineInstanceTemplateSpec{
+				Spec: cnv.VirtualMachineInstanceSpec{
+					Domain: cnv.DomainSpec{
+						Devices: cnv.Devices{
+							Interfaces: []cnv.Interface{
+								{
+									Name:       "net-0",
+									MacAddress: "00:50:56:be:b2:36",
+								},
+								{
+									Name:       "net-1",
+									MacAddress: "00:50:56:be:b2:37",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Simulate the builder's DeepCopy approach
+	targetVmSpec := sourceVm.Spec.DeepCopy()
```

Comment:
What is the point of these deepcopy tests? Contrary to what the comment says, these tests aren't testing anything the builder does; they're just testing that the DeepCopy method (which isn't MTV code) works.